edweiss-firebase/functions/src/functions/lectures/updateQuestion.ts

```diff
@@ -0,0 +1,29 @@
+import LectureDisplay from 'model/lectures/lectureDoc';
+import { onAuthentifiedCall } from 'utils/firebase';
+import { CollectionOf, getDocumentAndRef } from 'utils/firestore';
+import { fail, ok } from 'utils/status';
+
+export const updateQuestion = onAuthentifiedCall(LectureDisplay.Functions.updateQuestion, async (userId, args) => {
+    if (!args.id)
+        return fail("invalid_id");
+
+    const [question, ref] = await getDocumentAndRef(CollectionOf<LectureDisplay.Question>(`users/${userId}/lectures/${args.lectureId}/questions`), args.id);
+
+    if (question == undefined)
+        return fail("error_firebase");
+
+    const updatedQuestion: Partial<LectureDisplay.Question> = {
+        anonym: question.anonym,
+        likes: args.likes,
+        postedTime: question.postedTime,
+        text: question.text,
+        userID: question.userID
+    };
+    try {
+        await ref.update({ updatedQuestion });
```

Comment:
Why do you use ({}) and not just () for update function ?

Reply:
Braces are needed because Firestore’s update function requires an object as its argument. this could also be seen as { question: updatedQuestion }